sim-lib/src/lnd.rs

```diff
@@ -60,7 +62,27 @@ impl LightningNode for LndNode {
         unimplemented!()
     }
 
-    async fn track_payment(&self, _hash: PaymentHash) -> Result<(), LightningError> {
-        unimplemented!()
+    async fn track_payment(&self, hash: PaymentHash) -> Result<PaymentOutcome, LightningError> {
```

Comment:
Thinking about how we're going to use this, do you think it would make sense to pass a receiving channel in here which we report the payment outcome to? 

That way we can: 
1. Send the payment
2. Spin up a thread to track its outcome
3. Report the outcome to a single monitoring thread that writes them wherever we need them

Reply:
opted to skip streaming the result through a channel and instead refer a direct `PaymentResult`